api/api.go

```diff
@@ -318,6 +318,31 @@ func (api *api) UpdateApp(userApp *db.App, updateAppRequest *UpdateAppRequest) e
 }
 
 func (api *api) DeleteApp(userApp *db.App) error {
+	// Check if the app is a sub-wallet with a lightning address
+	var metadata map[string]interface{}
+	if userApp.Metadata != nil {
+		err := json.Unmarshal(userApp.Metadata, &metadata)
+		if err != nil {
+			logger.Logger.WithError(err).WithFields(logrus.Fields{
+				"app_id": userApp.ID,
+			}).Error("Failed to deserialize app metadata")
+		} else {
+			appStoreAppId, hasAppStoreId := metadata["app_store_app_id"]
+			lud16, hasLightningAddress := metadata["lud16"]
+
+			isSubwallet := hasAppStoreId && appStoreAppId == constants.SUBWALLET_APPSTORE_APP_ID
+
+			if isSubwallet && hasLightningAddress && lud16 != nil {
```

Comment:
All this is done to save a call to `GetAppById`, these checks are also done in `DeleteLightningAddress` again, should we remove this?